rust-runtime/smithy-types/src/lib.rs

```diff
@@ -3,26 +3,42 @@
  * SPDX-License-Identifier: Apache-2.0.
  */
 
+//! Protocol-agnostic types for smithy-rs.
+
+#![warn(
+    missing_docs,
+    missing_crate_level_docs,
+    missing_debug_implementations,
+    rust_2018_idioms,
+    unreachable_pub
+)]
+
+use std::collections::HashMap;
+
 pub mod base64;
 pub mod instant;
 pub mod primitive;
 pub mod retry;
 
-use std::collections::HashMap;
-
 pub use crate::instant::Instant;
 
+/// Binary Blob Type
+///
+/// Blobs represent protocol-agnostic binary content.
 #[derive(Debug, PartialEq, Clone)]
 pub struct Blob {
```

Comment:
This doesn't need to be added here, but what's the difference between a `Blob` and `Bytes`?

Reply:
Blob is just our newtype for binary data. We should probably change it to use Bytes internally

Reply:
Tracking in https://github.com/awslabs/smithy-rs/issues/785